### PR TITLE
Fix location issue

### PR DESCRIPTION
**Pull request by jichenjc**
fix this issue:
```
pkg/apis/ibmcloud/v1alpha1/zz_generated.deepcopy.go:278:15: in.Datacenter.DeepCopyInto undefined (type datatypes.Location has no field or method DeepCopyInto)
# sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/apis/ibmcloud/v1alpha1 [sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/apis/ibmcloud/v1alpha1.test]
pkg/apis/ibmcloud/v1alpha1/zz_generated.deepcopy.go:278:15: in.Datacenter.DeepCopyInto undefined (type datatypes.Location has no field or method DeepCopyInto)
```
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: config/crds/ibmcloud_v1alpha1_ibmcloudmachineproviderspecspec.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (gyliu513) [inline config/crds/ibmcloud_v1alpha1_ibmcloudmachineproviderspecspec.yaml]: should not this file name be `ibmcloud_v1alpha1_ibmcloudmachineproviderspec.yaml`?

BTW: How did you generate this file?

Comment (jichenjc) [reply config/crds/ibmcloud_v1alpha1_ibmcloudmachineproviderspecspec.yaml]: make file generate them , this is auto generated , I think your kubebuilder might have some issue so it lead to weird thing here like statusstatus and specspec definition
I made some changes according to openstack provider

# Generate manifests e.g. CRD, RBAC etc.
manifests:
        go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd
        kustomize build config/default/ > provider-components.yaml
        echo "---" >> provider-components.yaml
        kustomize build vendor/sigs.k8s.io/cluster-api/config/default/ >> provider-components.yaml

File: pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderstatus_types.go [hunk 2]
```diff
@@ -1,62 +0,0 @@
-/*
```
Comment (gyliu513) [inline pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderstatus_types.go]: why this file is being deleted? I think we still need this?

File: config/crds/ibmcloud_v1alpha1_ibmcloudmachineproviderstatus.yaml [hunk 3]
```diff
@@ -1,39 +0,0 @@
-apiVersion: apiextensions.k8s.io/v1beta1
-kind: CustomResourceDefinition
```
Comment (gyliu513) [inline config/crds/ibmcloud_v1alpha1_ibmcloudmachineproviderstatus.yaml]: Not clear why this file was deleted?

File: pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderstatus_types.go (hunk 2 above)
Comment (jichenjc) [reply pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderstatus_types.go]: openstack provider doesn't have this 

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/tree/master/config/crds
I thought maybe we don't need either

Comment (gyliu513) [reply pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderstatus_types.go]: Can we keep this for now, as this file was auto-generated by https://kubernetes-sigs.github.io/cluster-api/provider_implementations/generate_crds.html

Comment (jichenjc) [reply pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderstatus_types.go]: yes, I know seems the kubebuilder we used have some issue

The yaml actually generated from the pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineprovider*_types.go and we need adjust later

anyway, I restore the files

**Pull request merged**